react/features/conference/middleware.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+import {
+    CONFERENCE_JOINED,
+    VIDEO_QUALITY_LEVELS,
+    setReceiveVideoQuality
+} from '../base/conference';
+import { SET_REDUCED_UI } from '../base/reduced-ui/actionTypes';
+import { MiddlewareRegistry } from '../base/redux';
+import { setFilmstripVisible } from '../filmstrip';
+import { setToolboxVisible } from '../toolbox';
+
+MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
+    const result = next(action);
+
+    switch (action.type) {
+    case CONFERENCE_JOINED:
+    case SET_REDUCED_UI: {
+        const state = getState();
+        const { audioOnly } = state['features/base/conference'];
+        const { reducedUi } = state['features/base/reduced-ui'];
+
+        reducedUi && dispatch(setToolboxVisible(false));
```

Comment:
Here I'm commenting on "[RN] Adjust Conference for the reduced UI mode":

`setToolboxVisible` is an example of where my expectations about `visible` are shattered. One day we will merge with Web, the concept of reduced UI is really transferable to Web because it's built on top of the concept of a small enough size, here we're in a file which is common to Web and mobile and then we start hiding the toolbox in reduced-ui only to have it reappear when we move the mouse.

Anyway, please take my comments about `visible` and `enabled` as conversation topics.